python/pyspark/sql/pandas/serializers.py

```diff
@@ -385,37 +385,28 @@ def _create_struct_array(self, df, arrow_struct_type):
         """
         import pyarrow as pa
 
-        # Input partition and result pandas.DataFrame empty, make empty Arrays with struct
-        if len(df) == 0 and len(df.columns) == 0:
-            arrs_names = [
-                (pa.array([], type=field.type), field.name) for field in arrow_struct_type
-            ]
+        if len(df.columns) == 0:
+            return pa.array([{}] * len(df), arrow_struct_type)
```

Comment:
Under what scenario would it be possible for a DataFrame to have rows but no columns? i.e len(df.columns) == 0 but len(df) > 0?

Reply:
For example:

```py
>>> pdf = pd.DataFrame(index=[1,2,3])
>>> pdf
Empty DataFrame
Columns: []
Index: [1, 2, 3]
>>> len(pdf)
3
>>> len(pdf.columns)
0
```